### DISCUSSION TITLE
GPS problem

### DISCUSSION THREAD

**Discussion by Georgerom63** [iNavFlight/inav#7570] [Q&A]
I just build a drone for my needs actually for an insta360 one R and I put the gps as far as permitted by my design I shielded the cable but seems nothing change the behavior. Is trying to run away as soon I left the throttle. By the way, I did all the calibration away from any strong interferences. The toilet bowling is still there no matter what. I use Matek GPS & Compass Module - M8Q-5883 and FC SpeedyBee F7 Flight Controller Stack and I use the latest version of the Inav configurator. Don't know what else I can do to fix this annoying problem. Thanks, for any help
![20211024_171944](https://user-images.githubusercontent.com/93102737/138617471-ab366d9d-17c9-4b2d-9635-8777b4c1d85b.jpg)
![20211024_172012](https://user-images.githubusercontent.com/93102737/138617476-6965cb2e-a7d2-4cc7-80ae-85ce82465c13.jpg)
lp

**Comment by ryet9** [comment]
Any help?, well ok..
make sure **MAG Alignment** is **_CW 270° flip_**?
or umm TRY powering it by the **3V3** or **4V5** pads... if you are not already that is, 3V3 would be my suggestion, not the 5V as they kinda imply you should on the product page.. A side bonus is you can power the GPS/MAG via USB
😀

**Comment by Georgerom63** [comment]
> Any help?, well ok.. make sure **MAG Alignment** is **_CW 270° flip_**? or umm TRY powering it by the **3V3** or **4V5** pads... if you are not already that is, 3V3 would be my suggestion, not the 5V as they kinda imply you should on the product page.. A side bonus is you can power the GPS/MAG via USB 😀


I did connected to 4v5 pad and it is on CW270 flip. The thing I observed in Inav setup page  that image of the drone is moving allover the places. When I lift the nozzle of the drone up is also going to the right to a 90 degree and vice versa on the other side. Maybe I have a bad GPS module I really don't know what it cause this behavior.

**Comment by Pairan** [comment]
That thing with CW270° FLIP is no guarantee! I found some modules that differ here a lot! The only way to make sure that the alignment is correct is to double check it with (lets say) a smartphone ... should be almost moving and pointing in the same direction as the drone while showing N(orth) or 0°

so ... if the drone points to 0° ... does it really look to the north?

if not then there is a good chance that could be the reason for the toiletbowling and "strange" movements.

Further: 
only calibrate with the desired battery on! I found that the result really differed without it! => and that again results in toilet bowling and unwanted movements!

perhaps this helps

**Comment by rwindsorjr** [comment]
I have the same Maytek GPS/Compass. the Red dot on yours is on the front left corner. Mine is on the opposite corner  Right rear.  The wires and plug are plugged in the rear . The front socket is not used .   Hope that helps some !
Bob

**Comment by vlamgat007** [comment]
I have a couple of Matek M8Q's on wings and they work but they are not great.  I have one  Matek M10Q on another plane which is significantly better.  I also have a Foxeer M10Q on 10 inch quad with shielding.  The performance (#sats) drops significantly after powering up with the battery but it is still adequate.  I do not thing the weaker M8Q would work in my quad either, but I have not tested it.  There are many others that have made it work....it is just a bit more challenging.

**Reply by sensei-hacker** [reply]
Losing satellites after you power up indicates the VTX or another transmitter (telemetry?) is swamping the GPS. That's the difference between powering up the VTX vs not.

Mitigations include placing the GPS and VTX antennas at opposite ends of the aircraft and not using more VTX power than needed.
